src/main/kotlin/com/github/mitchellolsthoorn/testgenie/settings/TestGenieSettingsState.kt

```diff
@@ -3,5 +3,37 @@ package com.github.mitchellolsthoorn.testgenie.settings
 /**
  * This class is the actual data class that stores the values of the Settings entries.
  */
-data class TestGenieSettingsState
-    constructor(var globalTimeout: String = "60", var showCoverage: Boolean = false)
\ No newline at end of file
+data class TestGenieSettingsState(
+        var globalTimeout: String = "60",
+        var showCoverage: Boolean = false,
+
+        var sandbox : Boolean = true,
+        var assertions : Boolean = true,
+        var seed : String = "",
+        var algorithm: ContentDigestAlgorithm = ContentDigestAlgorithm.RANDOM_SEARCH,
+        var configurationId : String = "",
+        var clientOnThread : Boolean = false,
+        var junitCheck : Boolean = false,
+        var criterionLine : Boolean = true,
+        var criterionBranch : Boolean = true,
+        var criterionException : Boolean = true,
+        var criterionWeakMutation : Boolean = true,
+        var criterionOutput : Boolean = true,
+        var criterionMethod : Boolean = true,
+        var criterionMethodNoException : Boolean = true,
+        var criterionCBranch : Boolean = true,
+        var minimize : Boolean = true
+)
+
+enum class ContentDigestAlgorithm {
+    // random
+    RANDOM_SEARCH,
```

Comment:
Consider renaming the enums to display nice names like it is in the quick access tool window.

Reply:
We will discuss with the client which is their preferred way of displaying those parameters.